### DISCUSSION TITLE
How to send HEX data to SPI?

### DISCUSSION THREAD

**Discussion by swanduron** [micropython/micropython#11005] [General]
Hello experts,

I want to calculate (bool) and send the data to the SPI bus, but I note that the SPI only accepts certain data types bytes, or bytearray.

For example, I have below code slice:

```
CMD1 = const(64)
shiftAddress = CMD1 | offset
# spi.write(shiftAddress). << I want to do this but it doesn't work
```

Do anyone has the experience to convert HEX to bytes/bytearray in micropython?

Regards

**Comment by robert-hh** [comment]
You can do in on-the-fly be writing;
```
spi.write(bytearray([shiftAddress]))
```

or you can create a single element bytearray and store the value into it:
```
temp = bytearray(1)
temp[0} = shiftAddress
spi.write(temp)
```

**Reply by swanduron** [reply]
Thanks a LOT!!!!!!!!!!!!!
I just use a very clumsy method as below. Your ways look much better than I found.

```
temp = 0x12
spi.write(temp.to_bytes(1, 'big'))

```
Regards!
